### DISCUSSION TITLE
Long term support

### DISCUSSION THREAD

**Discussion by gilbertbw** [twinbasic/twinbasic#296] [General]
Hi @WaynePhillipsEA, This project/product looks very interesting. We have some very large VBA/MS Access projects (hundreds of thousands of LOC) that we have been considering how to modernise. twinBASIC presents a possible migration path without the massive cost of rewriting existing components in something like .NET. 

That said we have some concerns about twinBASIC becoming a core dependency of our products. twinBASIC could end up as critical to us as Windows and MS Access!

Our principle question is how confident are we that twinBASIC will still be around in ten or more years.

- What happens if you personally stop being able to support twinBASIC (win the lottery, retire etc.). What is the "bus factor" for the project?
- What happens if the company behind twinBASIC (iTech Masters?) ceases trading (sued, twinBASIC stops making commercial sense etc.)?

**Comment by WaynePhillipsEA** [comment]
Hi, and thanks for the question.

It's a legitimate concern, and one that I plan to address before or shortly after the first official release of twinBASIC.

The current plan is to place the full source code into an escrow service, or possibly under a similar agreement with a couple of trusted third parties.

Whatever happens, I am determined to ensure that twinBASIC has a vibrant future, and won't end up like VB6/VBA.  I will do whatever it takes to make sure that is the case.

Just for a little background on me; I've been in business since 2003, offering unique products and services for Microsoft Access (and VBA) over at EverythingAccess.com.  I run the business together with my partner Louise who handles most of the day to day admin work.   No plans for retirement any time soon 😉, but I completely understand the bus factor concerns.

Eventually I would like to see us partner with a larger firm which I feel would help with the overall "trust", but we may be a while off from that yet until tB matures and gets more recognition.

I'd be interested to hear from others on this issue.  If there are other things we can be doing to help ease the long-term support concerns, I'm eager to hear them.

**Reply by DavidHollman** [reply]
An interesting model to contemplate is what SQLite has done (https://www.sqlite.org/prosupport.html).  Its open source and public domain which seems like it will ensure continuity as long as people want to use it. But they also sell licenses for versions which contain certain proprietary features. I assume this balance has worked well for the SQLite developers and also seems to have been incredibly successful for SQLite as a product.

I'm just citing this as an example, not implying you should necessarily emulate it in every detail. Maybe there are other good models of mixed open+closed approaches to consider also.

At some point there was an attempt to get Microsoft to open source VB6; and all along there has been a vibrant VB6 community, and I think that effort & the interest in twinBASIC are reflective of that. So maybe there's a way to tap into that energy while also satisfying the very legitimate commercial interests.

**Reply by VB6-Programming** [reply]
Escrow is an approach to consider.  You can place your source code in escrow (and keep it updated).  Then in the event of twinBASIC no longer being supported/available (or Wayne losing an argument with a Boris bus 😉 ) the source code could be made open source.

**Comment by dmrvb** [comment]
A third and possibly more likely threat is that TB becomes reasonably successful and is purchased by a big software "investment" type company (making Wayne an offer he would be crazy to refuse) who then  totally wreck it, maybe by changing it out of all recognition to target a different market. There are many many examples of this.
I also worry about relying on a one person company but the big corporates don't really offer more security, they shut down successful projects for odd corporate reasons (an example....VB6 (how do I insert a smiley????)).
A good outcome would be for Wayne to expand to have 3 or 4 staff, all familiar with the code, but maybe Wayne would rather remain as a tiny one person company; employing and managing people is not nearly as much fun as writing software.
Some sort of open source sounds good but just making the code available is not enough, it needs somebody with the skills, inclination and time to take on the project at short notice should that be required.
I've been in a similar position to Wayne myself, there is no easy fix.

**Reply by KarlSevenSeven** [reply]
VB is not really ended, as it is still 'usable'.
We have a perpetual VB license, we can compile our apps and distribute them.

How is the possible future in case of the subscription model?
When you win the lottery, or any other event that makes the subscription suddenly no longer working...
...can we still use your software and do our business?

I assume that's not possible.
Please correct me.

**Reply by WaynePhillipsEA** [reply]
The subscription model was chosen in order to ensure the product is well maintained and has a long and vibrant future.  With the subscription model, it is in our interest to keep improving and maintaining the product so that our customers keep paying the subscription.  Nobody wants to pay for a perpetual licence and then have the product get no love or updates, which is what typically happens in my opinion.

As mentioned earlier in this discussion, we are looking to put the full source code into escrow, which means that if something goes horribly wrong at my end (bus factor), then the product is not dead at all because you'll be able to gain access to the source code.

**Reply by KarlSevenSeven** [reply]
While I don't like the subscription model too much, I clearly understand the reasons and **agree with you**.
A good decision for both parties!
Fine.

The source-in-escrow thing is fine as well.

**Reply by uwbwsvd** [reply]
In Germany I know "Verwantwortungseigentum" Stewardship? 
https://purpose-economy.org/en/. A Company of and for the people and not a ball capitalist can play with.

**Reply by dmrvb** [reply]
The UK has a community interest company or CIC, which should do good things for the community (which could be the user community?) and not give any money to shareholders. I think they also prevent the sale of company assets. Its just possible that TB could be moved to a CIC. Its much like a ltd company so no new admin skills to learn and has directors who can control the company hence avoiding the squabbling and forking of some open source projects. Not sure if directors can take a dividend but a CIC could employ Wayne with a suitable salary.
